x/bankplus/keeper/keeper.go

```diff
@@ -0,0 +1,176 @@
+package keeper
+
+import (
+	"context"
+
+	"cosmossdk.io/core/address"
+	"cosmossdk.io/core/store"
+	errorsmod "cosmossdk.io/errors"
+	"cosmossdk.io/log"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	"github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+var _ Keeper = (*BaseKeeper)(nil)
+
+type Keeper interface {
+	bankkeeper.Keeper
+
+	AddToInactiveAddr(ctx context.Context, address sdk.AccAddress)
+	DeleteFromInactiveAddr(ctx context.Context, address sdk.AccAddress)
+	IsInactiveAddr(address sdk.AccAddress) bool
+
+	InitializeBankPlus(ctx context.Context)
+}
+
+type BaseKeeper struct {
+	bankkeeper.BaseKeeper
+
+	ak      types.AccountKeeper
+	cdc     codec.Codec
+	addrCdc address.Codec
+
+	storeService   store.KVStoreService
+	inactiveAddrs  map[string]bool
+	deactMultiSend bool
+}
+
+func NewBaseKeeper(
+	cdc codec.Codec, storeService store.KVStoreService, ak types.AccountKeeper,
+	blockedAddr map[string]bool, deactMultiSend bool, authority string, logger log.Logger,
+) BaseKeeper {
+	return BaseKeeper{
+		BaseKeeper:     bankkeeper.NewBaseKeeper(cdc, storeService, ak, blockedAddr, authority, logger),
+		ak:             ak,
+		cdc:            cdc,
+		storeService:   storeService,
+		inactiveAddrs:  map[string]bool{},
+		deactMultiSend: deactMultiSend,
+		addrCdc:        cdc.InterfaceRegistry().SigningContext().AddressCodec(),
+	}
+}
+
+func (keeper BaseKeeper) InitializeBankPlus(ctx context.Context) {
+	keeper.loadAllInactiveAddrs(ctx)
+}
+
+// SendCoinsFromModuleToAccount transfers coins from a ModuleAccount to an AccAddress.
+// It will panic if the module account does not exist.
+func (keeper BaseKeeper) SendCoinsFromModuleToAccount(
+	ctx context.Context, senderModule string, recipientAddr sdk.AccAddress, amt sdk.Coins,
+) error {
+	senderAddr := keeper.ak.GetModuleAddress(senderModule)
+	if senderAddr.Empty() {
+		panic(errorsmod.Wrapf(sdkerrors.ErrUnknownAddress, "module account %s does not exist", senderModule))
+	}
+
+	if keeper.BlockedAddr(recipientAddr) {
+		return errorsmod.Wrapf(sdkerrors.ErrUnauthorized, "%s is not allowed to receive funds", recipientAddr)
+	}
+
+	return keeper.SendCoins(ctx, senderAddr, recipientAddr, amt)
```

Comment:
It would be better move the modification of `SendCoins()` to this method. It may also apply to other methods.